packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -31,42 +32,59 @@ class WeatherScreen extends HookConsumerWidget {
 
     final fullWeatherStateNotifier =
         ref.watch(w.fullWeatherStateNotifierProvider.notifier);
-    final controller = useFloatingSearchBarController();
 
-    final isLoading = useState(fullWeatherState is c.Loading);
+    final controller = useFloatingSearchBarController();
 
     final cityStateNotifier = ref.watch(c.cityStateNotifierProvider.notifier);
+    final cityState = ref.watch(c.cityStateNotifierProvider);
+
+    final refreshIndicatorKey = useGlobalKey<RefreshIndicatorState>();
 
-    Future<void> load() async {
-      await fullWeatherStateNotifier.loadFullWeather();
-    }
+    useEffect(
+      () {
+        Future.microtask(() => fullWeatherStateNotifier.loadFullWeather());
+        return null;
+      },
+      [fullWeatherStateNotifier],
+    );
 
     useEffect(
-      () => cityStateNotifier.addListener((state) {
-        if (state is c.Error) {
-          showFailureSnackBar(context, failure: state.failure, duration: 2);
+      () {
+        if (fullWeatherState.fullWeather == null) {
+          return null;
         }
-      }),
-      [cityStateNotifier],
+        return cityStateNotifier.addListener((state) {
+          if (state is c.Error) {
+            showFailureSnackBar(context, failure: state.failure, duration: 2);
+          }
+        });
+      },
+      [cityStateNotifier, fullWeatherState.fullWeather],
     );
 
     useEffect(
-      () => fullWeatherStateNotifier.addListener((state) {
-        if (state is w.Error) {
-          showFailureSnackBar(context, failure: state.failure, duration: 2);
+      () {
+        if (fullWeatherState.fullWeather == null) {
+          return null;
         }
-      }),
-      [fullWeatherStateNotifier],
+        return fullWeatherStateNotifier.addListener((state) {
+          if (state is w.Error) {
+            showFailureSnackBar(context, failure: state.failure, duration: 2);
+          }
+        });
+      },
+      [fullWeatherStateNotifier, fullWeatherState.fullWeather],
     );
 
     return Scaffold(
       resizeToAvoidBottomInset: false,
       body: FloatingSearchAppBar(
         liftOnScrollElevation: 0.0,
+        elevation: fullWeatherState.fullWeather == null ? 2.0 : 0.0,
```

Comment:
This elevation is present even with the banner, which looks a bit eh:
```suggestion
        // The elevation doesn't look great with the error banner.
        elevation: fullWeatherState.fullWeather == null && fullWeatherState is! w.Error ? 2.0 : 0.0,
```

Reply:
Actually, it looks weird without it, so nevermind.